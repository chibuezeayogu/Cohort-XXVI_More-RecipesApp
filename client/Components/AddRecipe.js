import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Input, Icon, Button, Col, Table} from 'react-materialize';
import axios from 'axios';
import Preloader from './Preloder';
//require('dotenv').config();


const CLOUDINARY_URL = process.env.CLOUDINARY_URL;
const CLOUDINARY_UPLOAD_PRESET = process.env.CLOUDINARY_UPLOAD_PRESET;


class AddRecipe extends Component{

    constructor(props) {
        super(props);
        this.state = {
            file: '',
            imagePreviewUrl: '', 
            ImageURL: '',
            arrIngredients:[],
            arrProcedures:[]
        };
      }

      componentWillMount(){
        // if(!this.props.userData.isAuthenticated){
        //     this.context.router.push('/sign');
        // }
        const token = localStorage.getItem('jwtToken');
        if(!token){
            this.context.router.push('/signin');
        }
    }
    //   componentWillUpdate(nextProps, nextState) {
    //     console.log(nextState.arrIngredients);
    // }
   
      _addIngredients(e){
          e.preventDefault();
          const ingredients = document.getElementById('ingredients').value;
          if(!ingredients){
                alert("Ingredients field is empty");
          }else {

            const newIngredients = this.state.arrIngredients.concat(ingredients);
                this.setState({
                    arrIngredients: newIngredients
                });
            }
      }

     



      _addProcedures(e){
        e.preventDefault();
        const procedures = document.getElementById('procedures').value;
        if(!procedures){
              alert("Ingredients field is empty");
        }else {

          const newProcedures = this.state.arrProcedures.concat(procedures);
              this.setState({
                    arrProcedures: newProcedures
              })
          }
        
          this.ingredient.state.reset();
      }

      _handleImageChange(e) {
        e.preventDefault();
    
        let reader = new FileReader();
        let file = e.target.files[0];
        const formData = new FormData;
        formData.append('file', file);
        formData.append('upload_preset', CLOUDINARY_UPLOAD_PRESET);
        
        // if (!file.type.match('image.*')){

        // }else{

        // }
        
        $.ajax({
            type: "POST",
            url: CLOUDINARY_URL,
            dataType: 'json',
            data: formData,
            contentType:false,
            processData: false,
            success: ((data) => {
                this.setState({ImageURL:data.secure_url});
            })
        });

        reader.onloadend = () => {
          this.setState({
            file: file,
            imagePreviewUrl: reader.result
          });
        };
    
        reader.readAsDataURL(file);
      }
      
    handleOnsubmit(e) {
        e.preventDefault();
        const title = this.title.state.value;
        const description = this.description.state.value;
        const ingredients = this.ingredients.state.value;
        const procedures = this.procedures.state.value;
        const imageURL = this.state.ImageURL;
        this.props.addRecipe(title, description, ingredients, procedures, imageURL);
        this.refs.AddRecipeForm.reset();
        this.setState({
            imagePreviewUrl:''
        });
    }
    render() {
        // let {imagePreviewUrl} = this.state;
        // let $imagePreview = null;
        // if (imagePreviewUrl) {
        //   $imagePreview =(<img  class="responsive-img circle" src={imagePreviewUrl}/>);
        // } else {
        //   $imagePreview = " ";
        // }
       const  { arrIngredients, arrProcedures } = this.state;
        return (
            <div className="container"> 
                <h1 className="center">Add Recipe</h1>
                <hr/>
                <form encType="multipart/form-data" methos="post" className="col l6 m8 s12 offset-l8 offset-m6" onSubmit={(e) => this.handleOnsubmit(e)} ref="AddRecipeForm">
                    <Row>
                        <Input s={12} label="Title"  type='text' ref={(input)=> {this.title=input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row s={8}>
                        <Input s={12} label="Description" className='validate' type='text' ref={(input) => {this.description = input}}><Icon>short_text</Icon></Input>
                    </Row>
                    <Row>
                        <Input s={12} label="Ingredients" id="ingredients" type='text' ref={(input) => {this.ingredients = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addIngredients(e)}}>Add Ingredient</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrIngredients
                                    .map((ingredient, i) => 
                                        <tr>
                                            <td className="s10">
                                                <input id="ingredients" type='text' value={ingredient}></input>
                                            </td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={12} label="Procedures" id="procedures" type='text' ref={(input) => {this.procedures = input}}><Icon>list</Icon></Input>
                        <Button waves='light' className='right' onClick={(e)=> {this._addProcedures(e)}}>Add Procedures</Button>
                    </Row>
                    <Row>
                        <Table className="striped">
                            <tbody>
                                {arrProcedures
                                    .map((procedures, i) => 
                                        <tr>
                                            <td className="s10">{procedures}</td>
                                            <td className="s1 right"><Icon>delete</Icon></td>
                                            <td className="s1 right"><Icon>edit</Icon></td>
                                        </tr>
                                    )
                                }
                            </tbody>
                        </Table>
                    </Row>
                    <Row>
                        <Input s={6} id="recipe"  name="file" type='file' ref={(input) => {this.file = input}}   onChange={(e)=>this._handleImageChange(e)}><Icon>insert_photo</Icon></Input>
                        <Col s={6} ><img className="responsive-img img-wh z-depth-3" src={this.state.imagePreviewUrl && this.state.ImageURL?this.state.imagePreviewUrl:"select image to preview"}/></Col>
                    </Row>
                    <br />
                    <Row>
                        <Button waves='light' className='right'><Icon right>send</Icon>Add</Button>
                    </Row>
                </form>
            </div>
        );
    }
}

AddRecipe.contextTypes = {
    router: React.PropTypes.object
};


AddRecipe.propTypes = {
handleOnSubmit: PropTypes.func.isRequired,
_addIngredients:  PropTypes.func.isRequired,
_addProcedures:  PropTypes.func.isRequired,
_handleImageChange:  PropTypes.func.isRequired,
addRecipe: PropTypes.func.isRequired,
};

export default AddRecipe;