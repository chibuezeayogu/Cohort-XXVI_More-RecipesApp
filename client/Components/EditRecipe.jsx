
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { connect } from 'react-redux';
import { editRecipe, getRecipe } from '../action/actionCreators';
import SmallPreloader from './SmallPreloader.jsx';
import Footer from './Footer/Footer.jsx';
import UserMenu from './Header/UserMenu.jsx';
import { validateEditRecipeForm, validateImage } from '../util/validateInputs';
import imageToFormData from '../util/ImageUpload';


/**
 * @description AddRecipe component for user to create a new recipe
 *
 * @class
 *
 * @extends Component
 *
 */
class EditRecipe extends Component {
   /**
   * @description initialize state and binds functiom
   *
   * @constructor
   *
   * @memberOf EditRecipe
   *
   * @returns {void}
   *
   */
  constructor() {
    super();
    this.state = {
      title: '',
      description: '',
      ingredients: '',
      procedures: '',
      imageUrl:'',
      image: {},
      selectedImage: '',
      errors: {},
      disabled: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleOnsubmit = this.handleOnsubmit.bind(this);
  }

   /**
   * @description routes to /recipes if user is authenticated
   *
   * @method
   *
   * @memberOf EditRecipe
   *
   * @returns {void}
   *
   */
  componentWillMount() {
    const token = localStorage.getItem('jwtToken');
    if (!token) {
      this.props.history.push('/signin');
    } else {
      const { id } = this.props.match.params;
      this.props.getRecipe(id);
    }
  }

  /**
   * @description checks if next recipes is fetched and disables is loading
   *
   * @method
   *
   * @memberOf SingleRecipe
   *
   * @param {Object} nextProps - nextProps object
   *
   * @returns {void}
   */
  componentWillReceiveProps(nextProps) {
    const { isFetched, isUpdated, recipes } = nextProps.userRecipeReducer;
    const { id } = this.props.match.params;
    if (isFetched) {
      const { id } = this.props.match.params;
     const index = recipes.findIndex(recipe => recipe.id === parseInt(id, 10));
      this.setState({ isLoading: false,
        title: recipes[index].title,
        description: recipes[index].description,
        ingredients: recipes[index].ingredients,
        procedures: recipes[index].procedures,
        imageUrl: recipes[index].imageUrl });
    } else if (isUpdated) {
      this.props.history.goBack();
    }
  }

  /**
   * @description get input and add to state
   *
   * @method
   *
   * @memberOf EditRecipe
   *
   * @param {Event} event - event object
   *
   * @returns {void}
   */
  handleChange(event) {
    event.preventDefault();
    this.setState({ errors: {} });
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @description get image input and add to state
   *
   * @method
   *
   * @memberOf EditRecipe
   *
   * @param {Event} event - event object
   *
   * @returns {void}
   */
  handleImageChange(event) {
    event.preventDefault();
    const file = event.target.files[0];
    this.setState({ errors: {}, image: {}, selectedImage: '' });
    this.setState({ image: file });
  
    const reader = new FileReader();
    reader.onload = () => {
        this.setState({ selectedImage: reader.result });
    }
    reader.readAsDataURL(file);
  }

  /**
   *
   * @description submits form
   *
   * @method
   *
   * @memberOf EditRecipe
   *
   * @param {Event} event - event object
   *
   * @returns {void}
   *
   */
  handleOnsubmit(event) {
    event.preventDefault();
    const err = validateEditRecipeForm(this.state);
    if (err.isError) {
      return this.setState({ errors: err.errors });
    }
    // console.log(Object.keys(this.state.image), 'with');
    // if(Object.keys(this.state.image)) {
    //   const uploadData = imageToFormData(this.state.image);
    //   delete axios.defaults.headers.common['Authorization'];
    //   axios(uploadData)
    //     .then((data) => {
    //     this.setState({ imageUrl: data.data.secure_url });
    //     console.log(this.state, 'with');
    //     this.props.editRecipe(this.state);
    //   }).catch((error) => { 
    //     this.setState({ disabled: false });
    //     Materialize.toast("Error! Please try again", 4000, 'red');
    //   });
    // } else {
    //   console.log(this.state, 'without');
    const { id } = this.props.match.params;
      this.props.editRecipe(id, this.state);
    // }
  }

  /**
   *
   * @description renders JSX element
   *
   * @method
   *
   * @memberOf SignUp
   *
   * @returns {void}
   *
   */
  render() {
    const { isFetched, recipes} = this.props.userRecipeReducer;
    
    if (this.state.isLoading) {
      return (
        <div className="body grey lighten-5">
            <UserMenu {...this.props} />
            <div className="main">
              <div className="container">
                <Preloader />
              </div>
            </div>
          <Footer />
          </div>
        )
      }
      
    if(isFetched === false && !recipes) {
      return (
        <div className="body grey lighten-5">
          <UserMenu {...this.props} />
          <div className="main">
            <div className="container">
              <h4 className="center-align" style={{ marginTop: 200 }}>
                No recipe found</h4>
            </div>
          </div>
        <Footer />
        </div>
      )
    }

    return (
      <div className="body grey lighten-5">
        <UserMenu />
        <div className="main">
          <div className="container">
            <div className="edit-recipe-form">
              <div className="col l6 m8 s12 offset-l6 offset-m4">
                <h4 className="center">Edit Recipe</h4>
                <hr />
              </div>
              <form onSubmit={this.handleOnsubmit}>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="title"
                      name="title"
                      type="text"
                      value={this.state.title}
                      onChange={this.handleChange}
                    />
                    <label htmlFor="title" className="active">Title</label>
                  </div>
                  <span className="right red-text error-margin">
                    {this.state.errors.titleError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <input
                      id="description"
                      name="description"
                      type="text"
                      value={this.state.description}
                      onChange={this.handleChange}
                    />
                    <label
                      htmlFor="description"
                      className="active"
                    >Description
                    </label>
                  </div>
                  <span className="right red-text error-margin">
                    {this.state.errors.descriptionError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="ingredients"
                      name="ingredients"
                      value={this.state.ingredients}
                      onChange={this.handleChange}
                      placeholder="enter ingredients seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text error-margin">
                    {this.state.errors.ingredientsError}
                  </span>
                </div>
                <div className="row">
                  <div className="input-field col s12">
                    <textarea
                      id="procedures"
                      name="procedures"
                      value={this.state.procedures}
                      onChange={this.handleChange}
                      placeholder="enter procedures seperated by semicolon(;)"
                    />
                  </div>
                  <span className="right red-text error-margin">
                    {this.state.errors.proceduresError}
                  </span>
                </div>
                <div className="row">
                  <div className="file-field input-field"
                    style={{ marginLeft: 10, marginRight: 10 }}>
                    <div className="btn green">
                      <span> 
                        <i 
                          className="material-icons"
                        >insert_photo
                        </i>
                      </span>
                      <input 
                        type="file" 
                        multiple 
                        onChange={event => this.handleImageChange(event)} />
                    </div>
                    <div className="file-path-wrapper">
                      <input 
                        className="file-path validate" 
                        type="text"
                        placeholder="Upload recipe image" />
                    </div>
                    <span className="right red-text error-margin">
                    {this.state.errors.imageError}
                  </span>
                  </div>
                  <div 
                    className="row s12 m4 l4 center-align" 
                    style={{ width: '100%'}}>
                    <img src={this.state.selectedImage||this.state.imageUrl} 
                      className="responsive-img center z-depth-2" 
                      style={{ maxHeight: 400, minHeight: 'auto', maxWidth: 400, minWidth: 'auto' }}/>
                  </div> 
                </div>
                <div className="row">
                  <button
                    className="btn right green"
                    type="submit"
                    name="action"
                    disabled={this.state.disabled}
                    style={{ marginRight: 10 }}
                  >Update
                  </button>
                </div>
                <hr />
                <div className="row center">
                  <div>
                    {this.state.disabled ?
                      <SmallPreloader />
                    :
                    ''
                    }
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

EditRecipe.propTypes = {
  editRecipe: PropTypes.func.isRequired,
  getRecipe: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired,
  userRecipeReducer: PropTypes.shape({
    isFetched: PropTypes.bool.isRequired
  }).isRequired
};

EditRecipe.contextTypes = {
  router: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  userRecipeReducer: state.userRecipeReducer,
});

export default connect(mapStateToProps, { editRecipe, getRecipe })(EditRecipe);

