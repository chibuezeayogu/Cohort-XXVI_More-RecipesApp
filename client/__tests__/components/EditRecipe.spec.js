import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import { EditRecipe } from '../../Components/EditRecipe';
import mockData from '../__mock__/actionMockData';

const props = {
  userRecipeReducer: {
    isFetched: false,
    isUpdated: false,
    recipes: []
  },
  history: {
    push: jest.fn(),
    goBack: jest.fn()
  },
  match: {
    params: {
      id: 1
    }
  },
  editRecipe: jest.fn(),
  fetchRecipe: jest.fn()
};

const event = {
  preventDefault: jest.fn(),
  target: {
    name: 'title',
    value: 'new title',
    files: [{}]
  }
};

const state = {
  id: '',
  title: '',
  description: '',
  ingredients: '',
  procedures: '',
  imageUrl: '',
  image: {},
  errors: {},
  disabled: false,
};

describe('<EditRecipe />', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<EditRecipe {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('<EditRecipe />', () => {
  it('should call handleChange function', () => {
    const wrapper = shallow(<EditRecipe {...props} {...state} />);
    wrapper.find('#title').simulate('change', event);
    expect(wrapper.instance().state.title).toEqual('new title');
  });
});

describe('HandleImageChange', () => {
  it('should set image state to user selected image', () => {
    const wrapper = shallow(<EditRecipe {...props} {...state} />);
    wrapper.find('#image').simulate('change', event);
    expect(wrapper.instance().state.image).toEqual({});
  });
});

describe('componentWillReceiveProps', () => {
  it('should call update component with new props', () => {
    const wrapper = shallow(<EditRecipe {...props} />);
    const componentWillReceivePropsSpy = jest
      .spyOn(wrapper.instance(), 'componentWillReceiveProps');
    wrapper.setProps({
      isFetched: true,
      isUpdated: true,
      recipes: mockData.recipe,
      match: { params: { id: 1 } },
    });
    expect(componentWillReceivePropsSpy).toHaveBeenCalled();
  });
});

describe('HandleOnSubmit', () => {
  it('should return error if input fields are empty', () => {
    const wrapper = shallow(<EditRecipe {...props} {...state} />);
    wrapper.find('form').simulate('submit', event);
    expect(wrapper.instance().state.errors.descriptionError)
      .toBe('description is required');
    expect(wrapper.instance().state.errors.ingredientsError)
      .toBe('ingredients is required');
    expect(wrapper.instance().state.errors.proceduresError)
      .toBe('procedures is required');
  });

  it('should return error for input did not meet the required lenght', () => {
    const wrapper = shallow(<EditRecipe {...props} {...state} />);
    wrapper.setState({
      title: 'abc',
      description: 'abc',
      ingredients: 'abc',
      procedures: 'abc',
    });
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(wrapper.instance().state.errors.titleError)
      .toBe('title must be at least 5 characters long');
    expect(wrapper.instance().state.errors.descriptionError)
      .toBe('description must be at least 5 characters long');
    expect(wrapper.instance().state.errors.ingredientsError)
      .toBe('ingredients must be at least 5 characters long');
    expect(wrapper.instance().state.errors.proceduresError)
      .toBe('procedures must be at least 5 characters long');
  });

  it('should sumbit form if all conditions are meet', () => {
    const wrapper = shallow(<EditRecipe {...props} {...state} />);
    wrapper.setState({
      title: 'recipes',
      description: 'awesome',
      ingredients: 'ingredients',
      procedures: 'procedures',
      image: { name: 'image', type: 'image/png' },
    });
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(wrapper.instance().state.disabled).toEqual(true);
  });
});
