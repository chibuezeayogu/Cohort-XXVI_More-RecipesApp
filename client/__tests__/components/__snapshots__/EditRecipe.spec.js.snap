// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditRecipe /> should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <div
    className="body grey lighten-5"
  >
    <withRouter(Connect(UserMenu)) />
    <div
      className="main"
    >
      <div
        className="container"
      >
        <div
          className="edit-recipe-form"
        >
          <div
            className="col l6 m8 s12 offset-l6 offset-m4"
          >
            <h4
              className="center"
            >
              Edit Recipe
            </h4>
            <hr />
          </div>
          <form
            onSubmit={[Function]}
          >
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <input
                  id="title"
                  name="title"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <label
                  className="active"
                  htmlFor="title"
                >
                  Title
                </label>
              </div>
              <span
                className="right red-text error-margin"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <input
                  id="description"
                  name="description"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <label
                  className="active"
                  htmlFor="description"
                >
                  Description
                </label>
              </div>
              <span
                className="right red-text error-margin"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <textarea
                  id="ingredients"
                  name="ingredients"
                  onChange={[Function]}
                  placeholder="enter ingredients seperated by semicolon(;)"
                  value=""
                />
              </div>
              <span
                className="right red-text error-margin"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="input-field col s12"
              >
                <textarea
                  id="procedures"
                  name="procedures"
                  onChange={[Function]}
                  placeholder="enter procedures seperated by semicolon(;)"
                  value=""
                />
              </div>
              <span
                className="right red-text error-margin"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="file-field input-field"
                style={
                  Object {
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
              >
                <div
                  className="btn green"
                >
                  <span>
                    <i
                      className="material-icons"
                    >
                      insert_photo
                    </i>
                  </span>
                  <input
                    id="image"
                    multiple={true}
                    onChange={[Function]}
                    type="file"
                  />
                </div>
                <div
                  className="file-path-wrapper"
                >
                  <input
                    className="file-path validate"
                    placeholder="Upload recipe image"
                    type="text"
                  />
                </div>
                <span
                  className="right red-text error-margin"
                />
              </div>
            </div>
            <div
              className="row"
            >
              <button
                className="btn right green"
                disabled={false}
                name="action"
                style={
                  Object {
                    "marginRight": 10,
                  }
                }
                type="submit"
              >
                Update
              </button>
            </div>
            <hr />
            <div
              className="row center"
            >
              <div>
                
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <Footer />
  </div>,
  "nodes": Array [
    <div
      className="body grey lighten-5"
    >
      <withRouter(Connect(UserMenu)) />
      <div
        className="main"
      >
        <div
          className="container"
        >
          <div
            className="edit-recipe-form"
          >
            <div
              className="col l6 m8 s12 offset-l6 offset-m4"
            >
              <h4
                className="center"
              >
                Edit Recipe
              </h4>
              <hr />
            </div>
            <form
              onSubmit={[Function]}
            >
              <div
                className="row"
              >
                <div
                  className="input-field col s12"
                >
                  <input
                    id="title"
                    name="title"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <label
                    className="active"
                    htmlFor="title"
                  >
                    Title
                  </label>
                </div>
                <span
                  className="right red-text error-margin"
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12"
                >
                  <input
                    id="description"
                    name="description"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                  <label
                    className="active"
                    htmlFor="description"
                  >
                    Description
                  </label>
                </div>
                <span
                  className="right red-text error-margin"
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12"
                >
                  <textarea
                    id="ingredients"
                    name="ingredients"
                    onChange={[Function]}
                    placeholder="enter ingredients seperated by semicolon(;)"
                    value=""
                  />
                </div>
                <span
                  className="right red-text error-margin"
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12"
                >
                  <textarea
                    id="procedures"
                    name="procedures"
                    onChange={[Function]}
                    placeholder="enter procedures seperated by semicolon(;)"
                    value=""
                  />
                </div>
                <span
                  className="right red-text error-margin"
                />
              </div>
              <div
                className="row"
              >
                <div
                  className="file-field input-field"
                  style={
                    Object {
                      "marginLeft": 10,
                      "marginRight": 10,
                    }
                  }
                >
                  <div
                    className="btn green"
                  >
                    <span>
                      <i
                        className="material-icons"
                      >
                        insert_photo
                      </i>
                    </span>
                    <input
                      id="image"
                      multiple={true}
                      onChange={[Function]}
                      type="file"
                    />
                  </div>
                  <div
                    className="file-path-wrapper"
                  >
                    <input
                      className="file-path validate"
                      placeholder="Upload recipe image"
                      type="text"
                    />
                  </div>
                  <span
                    className="right red-text error-margin"
                  />
                </div>
              </div>
              <div
                className="row"
              >
                <button
                  className="btn right green"
                  disabled={false}
                  name="action"
                  style={
                    Object {
                      "marginRight": 10,
                    }
                  }
                  type="submit"
                >
                  Update
                </button>
              </div>
              <hr />
              <div
                className="row center"
              >
                <div>
                  
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
      <Footer />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <EditRecipe
        editRecipe={[MockFunction]}
        fetchRecipe={
          [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          }
        }
        history={
          Object {
            "goBack": [MockFunction],
            "push": [MockFunction],
          }
        }
        match={
          Object {
            "params": Object {
              "id": 1,
            },
          }
        }
        userRecipeReducer={
          Object {
            "isFetched": false,
            "isUpdated": false,
            "recipes": Array [],
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": EditRecipe {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleChange": [Function],
        "handleOnsubmit": [Function],
        "props": Object {
          "editRecipe": [MockFunction],
          "fetchRecipe": [MockFunction] {
            "calls": Array [
              Array [
                1,
              ],
            ],
          },
          "history": Object {
            "goBack": [MockFunction],
            "push": [MockFunction],
          },
          "match": Object {
            "params": Object {
              "id": 1,
            },
          },
          "userRecipeReducer": Object {
            "isFetched": false,
            "isUpdated": false,
            "recipes": Array [],
          },
        },
        "refs": Object {},
        "state": Object {
          "description": "",
          "disabled": false,
          "errors": Object {},
          "id": "",
          "image": Object {},
          "imageUrl": "",
          "ingredients": "",
          "procedures": "",
          "title": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="body grey lighten-5"
        >
          <withRouter(Connect(UserMenu)) />
          <div
            className="main"
          >
            <div
              className="container"
            >
              <div
                className="edit-recipe-form"
              >
                <div
                  className="col l6 m8 s12 offset-l6 offset-m4"
                >
                  <h4
                    className="center"
                  >
                    Edit Recipe
                  </h4>
                  <hr />
                </div>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <input
                        id="title"
                        name="title"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <label
                        className="active"
                        htmlFor="title"
                      >
                        Title
                      </label>
                    </div>
                    <span
                      className="right red-text error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <input
                        id="description"
                        name="description"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <label
                        className="active"
                        htmlFor="description"
                      >
                        Description
                      </label>
                    </div>
                    <span
                      className="right red-text error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <textarea
                        id="ingredients"
                        name="ingredients"
                        onChange={[Function]}
                        placeholder="enter ingredients seperated by semicolon(;)"
                        value=""
                      />
                    </div>
                    <span
                      className="right red-text error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <textarea
                        id="procedures"
                        name="procedures"
                        onChange={[Function]}
                        placeholder="enter procedures seperated by semicolon(;)"
                        value=""
                      />
                    </div>
                    <span
                      className="right red-text error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="file-field input-field"
                      style={
                        Object {
                          "marginLeft": 10,
                          "marginRight": 10,
                        }
                      }
                    >
                      <div
                        className="btn green"
                      >
                        <span>
                          <i
                            className="material-icons"
                          >
                            insert_photo
                          </i>
                        </span>
                        <input
                          id="image"
                          multiple={true}
                          onChange={[Function]}
                          type="file"
                        />
                      </div>
                      <div
                        className="file-path-wrapper"
                      >
                        <input
                          className="file-path validate"
                          placeholder="Upload recipe image"
                          type="text"
                        />
                      </div>
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                  </div>
                  <div
                    className="row"
                  >
                    <button
                      className="btn right green"
                      disabled={false}
                      name="action"
                      style={
                        Object {
                          "marginRight": 10,
                        }
                      }
                      type="submit"
                    >
                      Update
                    </button>
                  </div>
                  <hr />
                  <div
                    className="row center"
                  >
                    <div>
                      
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
          <Footer />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="body grey lighten-5"
        >
          <withRouter(Connect(UserMenu)) />
          <div
            className="main"
          >
            <div
              className="container"
            >
              <div
                className="edit-recipe-form"
              >
                <div
                  className="col l6 m8 s12 offset-l6 offset-m4"
                >
                  <h4
                    className="center"
                  >
                    Edit Recipe
                  </h4>
                  <hr />
                </div>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <input
                        id="title"
                        name="title"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <label
                        className="active"
                        htmlFor="title"
                      >
                        Title
                      </label>
                    </div>
                    <span
                      className="right red-text error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <input
                        id="description"
                        name="description"
                        onChange={[Function]}
                        type="text"
                        value=""
                      />
                      <label
                        className="active"
                        htmlFor="description"
                      >
                        Description
                      </label>
                    </div>
                    <span
                      className="right red-text error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <textarea
                        id="ingredients"
                        name="ingredients"
                        onChange={[Function]}
                        placeholder="enter ingredients seperated by semicolon(;)"
                        value=""
                      />
                    </div>
                    <span
                      className="right red-text error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="input-field col s12"
                    >
                      <textarea
                        id="procedures"
                        name="procedures"
                        onChange={[Function]}
                        placeholder="enter procedures seperated by semicolon(;)"
                        value=""
                      />
                    </div>
                    <span
                      className="right red-text error-margin"
                    />
                  </div>
                  <div
                    className="row"
                  >
                    <div
                      className="file-field input-field"
                      style={
                        Object {
                          "marginLeft": 10,
                          "marginRight": 10,
                        }
                      }
                    >
                      <div
                        className="btn green"
                      >
                        <span>
                          <i
                            className="material-icons"
                          >
                            insert_photo
                          </i>
                        </span>
                        <input
                          id="image"
                          multiple={true}
                          onChange={[Function]}
                          type="file"
                        />
                      </div>
                      <div
                        className="file-path-wrapper"
                      >
                        <input
                          className="file-path validate"
                          placeholder="Upload recipe image"
                          type="text"
                        />
                      </div>
                      <span
                        className="right red-text error-margin"
                      />
                    </div>
                  </div>
                  <div
                    className="row"
                  >
                    <button
                      className="btn right green"
                      disabled={false}
                      name="action"
                      style={
                        Object {
                          "marginRight": 10,
                        }
                      }
                      type="submit"
                    >
                      Update
                    </button>
                  </div>
                  <hr />
                  <div
                    className="row center"
                  >
                    <div>
                      
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </div>
          <Footer />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <EditRecipe
    editRecipe={[MockFunction]}
    fetchRecipe={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
      }
    }
    history={
      Object {
        "goBack": [MockFunction],
        "push": [MockFunction],
      }
    }
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
    userRecipeReducer={
      Object {
        "isFetched": false,
        "isUpdated": false,
        "recipes": Array [],
      }
    }
  />,
}
`;
